base/errorshow.jl

```diff
@@ -165,30 +165,43 @@ end
 
 typesof(args...) = Tuple{Any[ Core.Typeof(a) for a in args ]...}
 
-function print_with_compare(io::IO, @nospecialize(a::DataType), @nospecialize(b::DataType), color::Symbol)
+struct ComparisonOptions
+    color::Symbol
+    constraining_typevars::Union{Nothing, Vector{TypeVar}}
```

Comment:
optional types as fields are generally frowned upon—it seems like this should just be an empty vector?